rust/datafusion/src/optimizer/projection_push_down.rs

```diff
@@ -114,6 +114,10 @@ fn optimize_plan(
     has_projection: bool,
 ) -> Result<LogicalPlan> {
     match plan {
+        LogicalPlan::Alias(plan, alias) => {
+            let new_input = optimize_plan(optimizer, &plan, &required_columns, true)?;
+            Ok(LogicalPlan::Alias(Box::new(new_input), alias.to_owned()))
```

Comment:
I suspect this code will eventually need a translation layer (from expressions in the sub plan to expressions in the outer query) -- I left more thoughts above